File: 1582. Special Positions in a Binary Matrix.py
class Solution:
    # two passes
    # O(mn) time and O(max(m, n)) space complexity
    def numSpecial(self, mat: List[List[int]]) -> int:
        r = [0] * len(mat)
        c = [0] * len(mat[0])

        for i, row in enumerate(mat):
            for j, num in enumerate(row):
                if num:
                    r[i] += 1
                    c[j] += 1
        
        ans = 0
        for i, row in enumerate(mat):
            for j, num in enumerate(row):
                if num and r[i] == 1 and c[j] == 1:
                    ans += 1

        return ans

File: 1582. Special Positions in a Binary Matrix.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        int r[mat.size()];
        for (int i = 0; i < mat.size(); i++) {
            r[i] = 0;
        };
        int c[mat[0].size()];
        for (int i = 0; i < mat[0].size(); i++) {
            c[i] = 0;
        };

        for (int i = 0; i < mat.size(); i++) {
            for (int j = 0; j < mat[0].size(); j++) {
                if (mat[i][j] == 1) {
                    r[i]++;
                    c[j]++;
                };
            };
        };

        int ans = 0;
        for (int i = 0; i < mat.size(); i++) {
            for (int j = 0; j < mat[0].size(); j++) {
                if (mat[i][j] == 1 && r[i] == 1 && c[j] == 1) {
                    ans++;
                };
            };
        };

        return ans;
    };
};
